src/main/resources/public/api-docs.yaml

```diff
@@ -214,108 +188,166 @@ paths:
               schema:
                 $ref: "#/components/schemas/SemanticAssetDetailsDto"
 components:
+  parameters:
+    limit:
+      name: limit
+      in: query
+      required: false
+      schema:
+        $ref: "#/components/schemas/Limit"
+    offset:
+      name: offset
+      in: query
+      required: false
+      schema:
+        $ref: "#/components/schemas/Offset"
+    agency_id:
+      name: agency_id
+      in: path
+      required: true
+      schema:
+        $ref: "#/components/schemas/UrlToken"
+    key_concept:
+      name: key_concept
+      in: path
+      description: |-
+        This parameter is an unique identifier for a controlled vocabulary.
+      required: true
+      schema:
+        $ref: "#/components/schemas/UrlToken"
+
   schemas:
+    URI:
+      description: |-
+        An utility schema for URI and URL.
+      type: string
+      format: url
+      maxLength: *max-url-size
+    Offset:
+      type: integer
+      format: int32
+      minimum: 0
+      maximum: *max-vocabulary-size
+      default: 0
+    Limit:
+      type: integer
+      format: int32
+      maximum: *max-limit-size
+      minimum: 1
+      default: 10
     VocabularyData:
       type: object
+      additionalProperties: false
       properties:
+        <<: *PaginatedResponse
         totalResults:
+          description: |-
+            TODO: The maximum number of entries in the collection
+            or the maximum returned entries?
           type: integer
           format: int64
-        limit:
-          type: integer
-          format: int32
-        offset:
-          type: integer
-          format: int64
+          minimum: 0
+          maximum: *max-vocabulary-size
         data:
           type: array
+          minItems: 0
+          maxItems: *max-limit-size
           items:
+            description: |-
+              A vocabulary item. Since
             type: object
+            maxProperties: 20
             additionalProperties:
               type: string
+              maxLength: 64
     NodeSummary:
       type: object
+      additionalProperties: false
       properties:
         iri:
-          type: string
+          $ref:  "#/components/schemas/URI"
         summary:
           type: string
+          maxLength: 120
     SearchResult:
       type: object
+      additionalProperties: false
       properties:
+        <<: *PaginatedResponse
         totalCount:
```

Comment:
this or totalResult https://github.com/teamdigitale/dati-semantic-backend/pull/23/files#diff-3a35ab406864494de60565273390f10d67ae481e505516e45ee8081a8648cf89L221 @ronytw